File: jaros/core/reasoning_boundary.py
"""The single Reasoning Plane boundary interface (EXT-001 / REQ-2).

All reasoning is funnelled through :class:`ReasoningBoundary`, whose only output
type is ``list[Decision]``. This module imports *only* from ``decision.py`` — it
holds no reference to the executor, state store, queues, or file system handles,
enforcing that nothing on the reasoning side can reach into execution directly.
"""

from __future__ import annotations

from typing import Protocol, runtime_checkable

from jaros.core.decision import Decision


# #EXT-001-REQ-2 Start
@runtime_checkable
class ReasoningBoundary(Protocol):
    """Interface every reasoning entry point is typed against.

    Its only output is inert ``Decision`` data; it cannot perform side effects.
    """

    def decide(self, context: object) -> list[Decision]:
        """Reason over ``context`` and return inert decisions only."""
        ...
# #EXT-001-REQ-2 End

File: src/core/reasoning-boundary.ts
// This module imports ONLY from decision.ts. It has no reference to the
// executor, state store, queue, or file system — the reasoning side cannot
// reach into execution.
import type { Decision } from "./decision";

// #EXT-001-REQ-2 Start
/**
 * The single seam through which all reasoning is funnelled. Its only output
 * type is `Decision` (a collection thereof). Implementations propose inert
 * decisions; they cannot perform side effects, mutate state, or invoke the
 * executor directly.
 */
export interface ReasoningBoundary {
  /**
   * Produce zero or more inert decisions for the given context. The `context`
   * is itself opaque to the boundary contract; implementations interpret it,
   * but may only ever return `Decision[]`.
   */
  decide(context: unknown): Promise<Decision[]>;
}
// #EXT-001-REQ-2 End
